include/boost/geometry/iterators/detail/segment_iterator/range_segment_iterator.hpp

```diff
@@ -100,23 +100,40 @@ class range_segment_iterator
             Reference
         >
 {
+private:
+    static bool has_less_than_two_elements(Range const& r)
+    {
+        iterator_type first = range_iterator_begin<Range>::apply(r);
+        iterator_type last = range_iterator_end<Range>::apply(r);
+
+        return first == last || ++first == last;
+    }
+
 public:
     typedef typename range_iterator_type<Range>::type iterator_type;
 
     // default constructor
     range_segment_iterator()
-        : m_it()
+        : m_it(), m_has_less_than_two_elements()
```

Comment:
I agree with initializing it with false.

I don't agree with the statement "I see it in various places, it is a general problem": those are assignments of setting a numeric type T to 0 (T()), which is perfectly fine and allowed. (as long as udt do support also this construct)